File: 3429-special-array-i/special-array-i.cpp
class Solution {
public:
    bool isArraySpecial(vector<int>& nums) {
        int i = 0 , j = 1;
        while(j<nums.size()){
            if((nums[i]%2 == 0 && nums[j]%2 != 0) || (nums[i]%2 != 0 && nums[j] %2 ==0)){
                
            }
            else{
                return false;
            }
            i+=1;
            j+=1;
        }
        return true;
    }
};

File: 3429-special-array-i/special-array-i.py
class Solution:
    def isArraySpecial(self, nums: List[int]) -> bool:
        if len(nums) ==1:
            return True

        i,j = 0,1
        while j < len(nums):
            if (nums[i]%2== 0 and nums[j]%2!=0) or (nums[i]%2 != 0 and nums[j]%2 == 0):
                None
            else:
                return False
            i,j = j,j+1
        return True